core/trino-main/src/main/java/io/trino/sql/SensitiveStatementRedactor.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql;
+
+import com.google.inject.Inject;
+import io.trino.FeaturesConfig;
+import io.trino.connector.CatalogFactory;
+import io.trino.spi.connector.ConnectorName;
+import io.trino.sql.tree.AstVisitor;
+import io.trino.sql.tree.CreateCatalog;
+import io.trino.sql.tree.Explain;
+import io.trino.sql.tree.ExplainAnalyze;
+import io.trino.sql.tree.Identifier;
+import io.trino.sql.tree.Node;
+import io.trino.sql.tree.Property;
+import io.trino.sql.tree.Statement;
+import io.trino.sql.tree.StringLiteral;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+
+public class SensitiveStatementRedactor
+{
+    public static final String REDACTED_VALUE = "***";
```

Comment:
We should consider a better value here than just `***`.  We could also consider using a special function like `$redacted$()`, which just throws exceptions if you try to actuall call that function.

Reply:
`***` seems to be almost what everyone uses for redaction.

Can you expand on the function idea? Is that to make it so that the output of `SHOW CREATE CATALOG` (as an example) is valid but still fails when you try to run it.